packages/notebook/src/browser/view-model/notebook-model.ts

```diff
@@ -205,21 +213,24 @@ export class NotebookModel implements Saveable, Disposable {
             this.dirtyCells.splice(this.dirtyCells.indexOf(cell), 1);
         }
 
-        const oldDirtyState = this._dirty;
-        this._dirty = this.dirtyCells.length > 0;
-        if (this.dirty !== oldDirtyState) {
-            this.onDirtyChangedEmitter.fire();
-        }
+        this.dirty = this.dirtyCells.length > 0;
     }
 
     setData(data: NotebookData): void {
         // Replace all cells in the model
         this.replaceCells(0, this.cells.length, data.cells, false);
         this.metadata = data.metadata;
-        this.dirty = false;
+        this.dirty = true;
```

Comment:
What's behind this change?

Reply:
It just aligns our behavior with VSCode. VSCode also sets notebooks as dirty if their data is overwritten programmatically.

Reply:
Hmh...looking at the call sites, this seems to be called from `revert`, for example. If it's a fix, maybe the fix needs to be at certain call sites, not here? What are the user-level scenarios that invoke this method?

Reply:
Probably, yeah, that's right. The caller site that I thought about would be [here](https://github.com/eclipse-theia/theia/blob/44b4f27c63028c1bdc7dcd9b565d666428af6f97/packages/plugin-ext/src/main/browser/notebooks/notebook-documents-main.ts#L165), but that's not the only one. I'll adjust this :)